gnowledge/gstudio
# Updating Node's discussion thread

After taking the pull, steps to be followed:
	1. python manage.py filldb
		* Creates new ATs: release_response, thread_interaction_type
	2. python manage.py sync_existing_documents
		* Updates all Page, File nodes' discussion thread.
		* If thread object exists, removes it from prior_node field of page/file node and creates a grelation 'has_thread' to link them and also creates above two defined AT with default values
		* Else creates a thread object and sets the gattributes and grelations
	3. python manage.py create_schema ATs.csv
		* Updates the ATs start_time and end_time by adding the 'Twist' in subject GST field

* While creating resource i.e Page and File a new tab is added 'Thread properties' which will be visible only to gstaff(superuser, group admin, group owner)

* Following 4 new attributes will be set with the discussion thread of Page/File node:
	[A].release_response - True/False
			default - True
			What to test:
                         1. Set True and check whether the replies are visible to all.
                         2. Set False and check whether all replies are visible to gstaff and others should be able to see only those replies which they have posted.

	[B].thread_interaction_type - basestring (a set of predefined values)
			default - Comment
			What to test:
                         1. Select any option and save. Check in the detail view of page,the button to discuss under the discussion tab should have the label as the selected option.

	[C].thread_start_time - datetime 
	
	[D].thread_end_time - datetime 
			* Comment or Reply button inside the replies is displayed only if current time is greater than or equal to thread start time or less than or equal to thread end time.

			What to test:
			 1. Set past date as start/end date. For time being, condition is relaxed to set min date (useful for testing). Go to detail view and discussion tab, the button will be hidden. Also, the Reply button inside the replies is also disabled.
			 2. Set start/end dates less than and greater than current date, respectively. You should be able to contribute to the discussion.

			Note : Dates attributes will be created only if both start and end date are entered.

* Templates modified:
	1. node_edit_base
		* Introduced 'Thread Properties' tab
	2. discussion.html
		* Capitalized Reply and Delete links inside a reply.
		* username inside a reply will link to userdashboard.
	3. UploadDoc.html
		* Added new fieldset for Thread properties while uploading a file.

New methods added to methods.py
	* create_thread_for_node
	* node_thread_access
